### PR TITLE
Fixes #5458 - use subnet dhcp boot mode if possible

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
warning: not tested most of it

**Comment by lzap** [comment]
I don't understand this patch, why you change only Debian part of this? Is Red Hat template already modified this way? Can you explain please? Is 61cfd65bc9c96ee483c38fbc578bf05c4e849faa related?

**Comment by domcleal** [comment]
@lzap yes, the commit you linked changed the kickstart to obey the subnet boot mode and not just the (deprecated?) `@static` variable.

File: preseed/provision.erb [hunk 1]
```diff
@@ -146,4 +152,4 @@ d-i grub-installer/bootdev string default
 <% end -%>
 d-i finish-install/reboot_in_progress note
 
-d-i preseed/late_command string wget -Y off <%= @static ? "'#{foreman_url('finish')}&static=true'" : foreman_url('finish') %> -O /target/tmp/finish.sh && in-target chmod +x /tmp/finish.sh && in-target /tmp/finish.sh
+d-i preseed/late_command string wget -Y off <%= dhcp ? foreman_url('finish') : "'#{foreman_url('finish')}&static=true'" %> -O /target/tmp/finish.sh && in-target chmod +x /tmp/finish.sh && in-target /tmp/finish.sh
```
Comment (domcleal) [inline preseed/provision.erb]: Perhaps keep the conditional based on `@static` here, so we don't begin passing static=yes to finish scripts when a subnet is set to static.  Finish scripts can still check for the subnet boot mode.

**Comment by domcleal** [comment]
I haven't had chance to test this, but would like to see it in 1.8 for parity with kickstarts - if anybody gets the chance to install via preseed to see that the subnet boot mode on static configures the host correctly (or at least attempts to), it'd be appreciated.

**Comment by mmoll** [comment]
tested, working fine :+1:

File: preseed/provision.erb (hunk 1 above)
Comment (ares) [inline preseed/provision.erb]: Makes sense, reverted.

**Comment by ares** [comment]
The only change I did after @mmoll testing was reverting the condition in preseed provision template as @domcleal suggested.

**Comment by domcleal** [comment]
Thanks @ares and @mmoll, merged as 5339d8c.

**Pull request closed without merging**